src/cryptoadvance/specter/util/bitcoind_setup_tasks.py

```diff
@@ -0,0 +1,212 @@
+import os, time, requests, secrets, platform, tarfile, zipfile, sys
+from ..bitcoind import BitcoindPlainController
+import pgpy
+from pathlib import Path
+from .sha256sum import sha256sum
+
+
+def setup_bitcoind_thread(specter=None):
+    try:
+        BITCOIND_OS_SUFFIX = {
+            "Windows": "win64.zip",
+            "Linux": "x86_64-linux-gnu.tar.gz",
+            "Darwin": "osx64.tar.gz",
+        }
+        bitcoind_url = f"https://bitcoincore.org/bin/bitcoin-core-0.21.0/bitcoin-0.21.0-{BITCOIND_OS_SUFFIX[platform.system()]}"
+        response = specter.requests_session().get(bitcoind_url, stream=True)
```

Comment:
Here for the first time we're downloading a file and store it in the data-folder. This will happen another 3 times, 2 of them not tracked, one tracked as well. Can't we refactor that out to a an own function ? 
My guess it will simplify the code and probably also make it shorter?